releng_sop/pulp_clone_repos.py

```diff
@@ -0,0 +1,365 @@
+        :rtype:  list of strings
+        """
+        # password is added only if requested and this is commit action
+        password = []
+        if add_password and commit:
+            password = ["--password=%s" % self.pulp_password]
+
+        echo = []
+        if not commit:
+            echo = ['echo']
+
+        commands = []
+        for repo in self.cloned:
+            cmd = []
+            cmd.append("pulp-admin")
+            cmd.append("--config=%s" % self.pulp_config.config_path)
+            cmd.append("--user=%s" % self.pulp_config["client"]["user"])
+            cmd = cmd + password
+            cmd.append("repo")
+            cmd.append("clone")
+            cmd.append("--id=%s" % repo['from'])
+            cmd.append("--clone_id=%s" % repo['to'])
+            cmd = echo + cmd
+            commands.append(cmd)
+        return commands
+
+    def run(self, commit=False):
+        """Print command details, get command and run it."""
+        details = self.details(commit=commit)
+        print(details)
+        commands_exec = self.get_cmd(add_password=True, commit=commit)
+        commands_print = self.get_cmd(add_password=False, commit=commit)
+        for cmd_exec, cmd_print in zip(commands_exec, commands_print):
+            print(cmd_print)
+            subprocess.check_call(cmd_exec)
+        if self.sameName:
+            for x in self.sameName:
+                print('Source and destination is the same. Cloning "%s" skipped.' % x['from'])
+        if self.missDest:
+            for x in self.missDest:
+                print('Missing destination repo. Cloning from "%s" skipped.' % x['to'])
+        if self.missSource:
+            for x in self.missSource:
+                print('Missing source repo. Cloning from "%s" skipped.' % x['to'])
+
+
+def get_parser():
+    """Construct argument parser.
+
+    :returns: ArgumentParser object with arguments set up.
+    :rtype: argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "from_release_id",
+        metavar="FROM_RELEASE_ID",
+        help="PDC release ID to clone from, for example 'fedora-24', 'fedora-24-updates'.",
+    )
+    parser.add_argument(
+        "to_release_id",
+        metavar="TO_RELEASE_ID",
+        help="PDC release ID to clone to, for example 'fedora-24', 'fedora-24-updates'.",
+    )
+    parser.add_argument(
+        "repo_family",
+        metavar="REPO_FAMILY",
+        help="PDC repo family to be considered for cloning.",
+    )
+    parser.add_argument(
+        "--variant",
+        metavar="VARIANT",
+        dest="variants",
+        action="append",
+        help="PDC variant(s) to be considered for cloning.",
+    )
+    parser.add_argument(
+        "--arch",
+        metavar="ARCH",
+        dest="arches",
+        action="append",
+        help="PDC architecture(s) to be considered for cloning.",
+    )
+    parser.add_argument(
+        "--content-category",
```

Comment:
Put this way, `--content-category` will be a required option, which should not be the case. It should work like `--arch` and `--variant` does.